src/monster.cpp

```diff
@@ -1992,12 +1992,12 @@ void monster::drop_items_on_death()
         return;
     }
     const auto dropped = g->m.put_items_from_loc( type->death_drops, pos(), calendar::turn );
-    if( !type->in_species( ZOMBIE ) && !type->in_species( FUNGUS ) ) {
-        return;
-    }
-    for( const auto &it : dropped ) {
-        if( it->is_armor() ) {
-            it->item_tags.insert( "FILTHY" );
+
+    if( type->in_species( ZOMBIE ) || type->in_species( FUNGUS ) || has_flag( MF_FILTHY ) ) {
```

Comment:
Ideally this PR would remove the species checks entirely and just add this tag to all zombies, in case we want to have some zombies drop clean clothes (If I _had_ to give an example, perhaps the clothes on hazmat zombies would be clean, but that's just an example and I'm not suggesting we actually do that.)

Reply:
See issue #26228

Reply:
Have added NOT_FILTY tag.